Google.Api.Gax.Grpc/ClientStreamingBase.cs

```diff
@@ -0,0 +1,116 @@
+        /// <returns>A <see cref="Task"/> which will complete when the message has been written to the stream.</returns>
+        public virtual Task WriteAsync(TRequest message)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Writes a message to the stream, if there is enough space in the buffer and <see cref="WriteCompleteAsync"/>
+        /// hasn't already been called.
+        /// </summary>
+        /// <param name="message">The message to write.</param>
+        /// <param name="options">The write options to use for this message.</param>
+        /// <returns><c>null</c> if the message queue is full or the stream has already been completed.</returns>
+        public virtual Task TryWriteAsync(TRequest message, WriteOptions options)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Writes a message to the stream, if there is enough space in the buffer and <see cref="WriteCompleteAsync"/>
+        /// hasn't already been called.
+        /// </summary>
+        /// <param name="message">The message to write.</param>
+        /// <param name="options">The write options to use for this message.</param>
+        /// <exception cref="InvalidOperationException">There isn't enough space left in the buffer,
+        /// or <see cref="WriteCompleteAsync"/> has already been called.</exception>
+        /// <returns>A <see cref="Task"/> which will complete when the message has been written to the stream.</returns>
+        public virtual Task WriteAsync(TRequest message, WriteOptions options)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Completes the stream when all buffered messages have been sent.
+        /// Only the first call to this method on any instance will have any effect;
+        /// subsequent calls will return <c>null</c>.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> which will complete when the stream has finished being completed;
+        /// or <c>null</c> if this method has already been called.</returns>
+        public virtual Task TryWriteCompleteAsync()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Completes the stream when all buffered messages have been sent. This method can only be called
+        /// once, and further messages cannot be written after it has been called.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">This method has already been called.</exception>
+        /// <returns>A <see cref="Task"/> which will complete when the stream has finished being completed.</returns>
+        public virtual Task WriteCompleteAsync()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Asyncronous call result
```

Comment:
Possibly expand this to clarify that it's expected that this will only complete successfully at some point after the client has called `WriteCompleteAsync`?

Reply:
👍